symphony-bdk-http/symphony-bdk-http-api/src/main/java/com/symphony/bdk/http/api/HttpClient.java

```diff
@@ -0,0 +1,241 @@
+package com.symphony.bdk.http.api;
+
+import com.symphony.bdk.http.api.util.GenericClass;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * HttpClient is the main entry point to fluent API used to build an {@link ApiClient} and execute
+ * client requests in order to consume responses returned
+ */
+public class HttpClient {
```

Comment:
if we want to enforce the use of the builder we can add a private constructor here

Reply:
I only added the constructor for unittest purpose. Jacoco complained that the code coverage on HttpClient didn't reach 90% even when intellij said 100%, I think it's because the default constructor is not tested (not really sure about it)